pkg/aws/eni/node.go

```diff
@@ -369,10 +369,10 @@ func (n *Node) allocateENI(s *types.Subnet, a *allocatableResources) error {
 
 	scopedLog.Info("Attached ENI to instance")
 
-	if nodeResource.Spec.ENI.DeleteOnTermination {
+	if *nodeResource.Spec.ENI.DeleteOnTermination {
```

Comment:
Shouldn't we check if `nodeResource.Spec.ENI.DeleteOnTermination != nil` first?

Reply:
It can't be nil because of what we do during initialization:
https://github.com/cilium/cilium/pull/9406/files/b3b0cb22bdaf9b22557f8842c4f7220be31ddfb4#diff-3f32d71dfbe1cd9d0fa51b5ed92c1366R307

The idea is that `nodeResource.Spec.ENI.DeleteOnTermination` is always true or false but `c.ENI.DeleteOnTermination` can be true, false or nil

We could still add the check though (and default to true?)

Reply:
Probably we should leave the line 307 alone and perform the nil check here.

Reply:
Sounds good to me, this is simpler